### PR TITLE
Milestone 4: Connect API to redux 

### PR DESCRIPTION
**Pull request by zdnahom**
## Milestone 4: Connect API to redux 
In this milestone I :
- Fetch data from BookstoreAPI (using axios and createAsyncThunk)
- Refactored Add book functionality
- Refactored Remove book functionality

### CODE REVIEW

File: src/components/Book.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-    <li>
+    <li key={bookInfo.item_id}>
       <div className="book-content">
         <p>{bookInfo.title}</p>
         <span>{bookInfo.author}</span>
-        <button type="button" onClick={() => dispatch(removeBook(bookInfo.item_id))}>Remove</button>
+        <button type="button" onClick={() => dispatch(deleteBook(bookInfo.item_id))}>Remove</button>
       </div>
     </li>
   );
```
Comment (Zuhaib042) [inline src/components/Book.js]: - [ ] Your logic is on point but there is something wrong with the rendering. sometimes the new books `li` are added to the book list `ul` and sometimes the author and title don't appear. only the remove button appears. and if you add 3 or 4 books like that and then click on any remove button, then all the `li`s gets removed. you can see this in the following video. kindly consider fixing this bug.

https://user-images.githubusercontent.com/96683843/230124691-864f0e3b-2974-4892-b1d7-4538dc5d4cf2.mov

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @zdnahom , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Book.js (hunk 1 above)
Comment (zdnahom) [reply src/components/Book.js]: Thanks so much for your insights @Zuhaib042, I've fixed the bug. 🙏

**Review by frank1738 — APPROVED** [review]
Hi @zdnahom , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Gitflow
✅ No linter error
✅ Good documentation

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**